### PR TITLE
Fix issue #159 [Use non-parameter constructor of DeviceTwinDevice if device id is null]

### PR DESCRIPTION
**Pull request by zhang-hua**
When scheduling a twin job, the job result contains empty device id which
will fail the constructor of DeviceTwinDevice(String deviceId). The
non-parameter constructor should be used in this case.

 

# Checklist
- [x] I have read the [contribution guidelines] (https://github.com/Azure/azure-iot-sdk-java/blob/master/.github/CONTRIBUTING.md).
- [x] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- If this is a modification that impacts the behavior of a public API
  - [ ] I edited the corresponding document in the `devdoc` folder and added or modified requirements.
- I submitted this PR against the correct branch: 
  - [x] This pull-request is submitted against the `master` branch. 

# Reference/Link to the issue solved with this PR (if any)
https://github.com/Azure/azure-iot-sdk-java/issues/159

# Description of the problem

When scheduling a twin job, the job result contains null or empty device id which
will fail the constructor of DeviceTwinDevice(String deviceId). The
non-parameter constructor should be used in this case.
# Description of the solution

### CODE REVIEW

File: service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/jobs/JobResult.java [hunk 1]
```diff
@@ -155,7 +155,8 @@
         TwinParser twinParser = jobsResponseParser.getUpdateTwin();
         if(twinParser != null)
         {
-            this.updateTwin = new DeviceTwinDevice(twinParser.getDeviceId());
+            this.updateTwin = twinParser.getDeviceId() == null || twinParser.getDeviceId().isEmpty() ?
+                new DeviceTwinDevice() : new DeviceTwinDevice(twinParser.getDeviceId());
```
Comment (prmathur-microsoft) [inline service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/jobs/JobResult.java]: Please add a test case for empty device ID in the corresponding test.

Comment (prmathur-microsoft) [reply service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/jobs/JobResult.java]: Test suite for this is located https://github.com/Azure/azure-iot-sdk-java/blob/master/service/iot-service-client/src/test/java/tests/unit/com/microsoft/azure/sdk/iot/service/jobs/JobResultTest.java

Comment (zhang-hua) [reply service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/jobs/JobResult.java]: @prmathur-microsoft , a test case of using nullable deviceId has been added in the commit. please help review it. Thanks!

**Comment by prmathur-microsoft** [comment]
This PR has been merged in master. Thanks for contributing!
